src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUObject.java

```diff
@@ -69,67 +76,69 @@ public boolean isAllocated() {
 	 * Then returns toBeRemoved. 
 	 * 
 	 */
-	protected void evict(long GPUSize) throws DMLRuntimeException {
-		if(GPUContext.allocatedPointers.size() == 0) {
-			throw new DMLRuntimeException("There is not enough memory on device for this matrix!");
-		}
-		
-		Statistics.cudaEvictionCount.addAndGet(1);
-		
-		synchronized(evictionLock) {
-			Collections.sort(GPUContext.allocatedPointers, new Comparator<GPUObject>() {
-	
-				@Override
-				public int compare(GPUObject p1, GPUObject p2) {
-					int p1Val = p1.numLocks.get();
-					int p2Val = p2.numLocks.get();
-					
-					if(p1Val < 0 || p2Val < 0) {
-						throw new RuntimeException("Number of locks cannot be negative");
-					}
-					else if(p1Val == 0 && p2Val == 0) {
-						// Both p1 and p2 are unlocked, return largest object
-						// TODO: Modify this !!
-						long p1Size = 0; long p2Size = 0;
-						try {
-							p1Size = p1.getSizeOnDevice();
-							p2Size = p2.getSizeOnDevice();
-						} catch (DMLRuntimeException e) {
-							throw new RuntimeException(e);
-						}
-						if(p1Size == p2Size) {
-							return 0;
-						}
-						else if(p1Size < p2Size) {
-							return 1;
-						}
-						else {
-							return -1;
-						}
-					}
-					else if(p1Val > p2Val) {
-						// There are more locks on p1
-						return 1;
-					}
-					else {
-						// There are more locks on p2
-						return -1;
-					}
-				}
-			});
-			
-			
-			while(GPUSize > getAvailableMemory() && GPUContext.allocatedPointers.size() > 0) {
-				GPUObject toBeRemoved = GPUContext.allocatedPointers.get(GPUContext.allocatedPointers.size() - 1);
-				if(toBeRemoved.numLocks.get() != 0) {
-					throw new DMLRuntimeException("There is not enough memory on device for this matrix!");
-				}
-				if(toBeRemoved.isDeviceCopyModified) {
-					toBeRemoved.copyFromDeviceToHost();
-				}
-				toBeRemoved.clearData();
-			}
-		}
+	protected void evict(final long GPUSize) throws DMLRuntimeException {
```

Comment:
This method seems to have been changed by you. Since you've figured out what it does, could you please add comments?